master/buildbot/util/latent.py

```diff
@@ -40,6 +40,9 @@ def renderWorkerPropsOnStart(self, build):
         self._actual_build_props = copy.deepcopy(props)
         defer.returnValue(props)
 
+    def resetWorkerPropsOnStop(self):
+        self._actual_build_props = None
```

Comment:
I guess https://github.com/buildbot/buildbot/pull/5544 was actually trying to avoid rescheduling on the same worker if the properties were incompatible. Is this an overlook in #5544 ?

I would avoid calling this in all subclass, but rather generically reset the _actual_build_props in the generic code.

Reply:
I'm not sure where in the generic class `CompatibleLatentWorkerMixin` this could be done? There is no hook whatsoever that calls back into the `CompatibleLatentWorkerMixin` when the instance is destroyed.

Or do you mean that `AbstractLatentWorker` should call `CompatibleLatentWorkerMixin.resetWorkerPropsOnStop()`?
That would violate that `AbstractLatentWorker` isn't necessarily a `CompatibleLatentWorkerMixin`, wouldn't it?